### PR TITLE
DB-8394 Remove extra wait for Spark queries

### PR DESCRIPTION
**Pull request by dgomezferro**

### CODE REVIEW

File: splice_machine/src/main/java/com/splicemachine/derby/stream/function/MergeAllAggregatesFunction.java [hunk 1]
```diff
@@ -43,7 +43,6 @@ public ExecRow call(ExecRow locatedRow1, ExecRow locatedRow2) throws Exception {
             aggregates = op.aggregates;
             initialized = true;
         }
-        operationContext.recordRead();
```
Comment (yxia92) [inline splice_machine/src/main/java/com/splicemachine/derby/stream/function/MergeAllAggregatesFunction.java]: Why is this line removed?

File: hbase_sql/src/main/java/com/splicemachine/stream/RemoteQueryClientImpl.java [hunk 2]
```diff
@@ -174,16 +174,16 @@ private void updateLimitOffset() throws StandardException {
...
-            olapFuture.get(time, unit);
+            olapFuture.cancel(true);
     }
 
     @Override
     public void close() throws Exception {
         streamListener.stopAllStreams();
         if (olapFuture != null)
-            olapFuture.cancel(true);
+            olapFuture.cancel(false);
```
Comment (yxia92) [inline hbase_sql/src/main/java/com/splicemachine/stream/RemoteQueryClientImpl.java]: Interrupt() and close() seem to be the same, as the parameter true or false for the cancel() function is not used. Is it intended to keep two separate functions so that in the future different operations will be applied?

File: splice_machine/src/main/java/com/splicemachine/derby/stream/function/MergeAllAggregatesFunction.java (hunk 1 above)
Comment (dgomezferro) [reply splice_machine/src/main/java/com/splicemachine/derby/stream/function/MergeAllAggregatesFunction.java]: It's an unrelated change, we were overcounting rows produced on GroupedAggregattes. I'd rather leave it in since it's a very simple change.

File: hbase_sql/src/main/java/com/splicemachine/stream/RemoteQueryClientImpl.java (hunk 2 above)
Comment (dgomezferro) [reply hbase_sql/src/main/java/com/splicemachine/stream/RemoteQueryClientImpl.java]: You are right, I thought we were handling it in OlapFuture.cancel() and my limited testing didn't reproduce the exception from SPLICE-1895  

I addressed it now by ignoring the Cancel message when we cancel without interrupt.

File: splice_machine/src/main/java/com/splicemachine/derby/stream/function/MergeAllAggregatesFunction.java (hunk 1 above)
Comment (yxia92) [reply splice_machine/src/main/java/com/splicemachine/derby/stream/function/MergeAllAggregatesFunction.java]: sure, thanks!

File: hbase_sql/src/main/java/com/splicemachine/stream/RemoteQueryClientImpl.java (hunk 2 above)
Comment (yxia92) [reply hbase_sql/src/main/java/com/splicemachine/stream/RemoteQueryClientImpl.java]: Thanks. Now that remoteQueryClient.close() is used for successful execution, interrupt() is used for timeout or killed scenario, should we change the call of remoteQueryClient.close() in SpliceBaseOperation.kill() and timeout() to interrupt() instead?

Comment (dgomezferro) [reply hbase_sql/src/main/java/com/splicemachine/stream/RemoteQueryClientImpl.java]: Yep, good catch. Thanks for the review!
